src/common-model/functions.ts

```diff
@@ -218,6 +224,39 @@ export function getSampleRegistrationDataFromDonor(donor: DeepReadonly<Donor>) {
   return sample_registration;
 }
 
+export const dnaSampleFilter = (specimen: Specimen): boolean =>
+  // All Specimen Have a Sample Registration
+  // Only DNA records count towards completion
+  specimen.samples.filter(sample => dnaSampleTypes.includes(sample.sampleType)).length > 0;
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some
work here?

Reply:
Yes it does